src/components/ReportsPage/AllScorecardsPage/HeatmapFiltersSelect.tsx

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2024 Cortex Applications, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Box, Button, FormControl, InputLabel, MenuItem, Select } from '@material-ui/core';
+import { Clear } from '@material-ui/icons';
+import React from 'react';
+
+export interface ModalSelectItem {
+  value: string | number;
+  label: string;
+}
+
+interface ModalSelectProps<T> {
+  name: string,
+  value: T[],
+  onChange: (value: T[]) => void,
+  onReset: () => void,
+  options: ModalSelectItem[]
+}
+
+export const ModalSelect = <T,>({ name, value, onChange, onReset, options }: ModalSelectProps<T>) => {
```

Comment:
Looks like we added generics and then never used it?

Reply:
I have just moved this file to a new destination, because the `AllScorecardPage` still requires this file.. but there's a casting on line 41 and I would say that's the reason why the generics were used here.